frontends/mit-open/src/page-components/LearningResourceDrawer/LearningResourceDrawer.tsx

```diff
@@ -27,14 +28,38 @@ const PAPER_PROPS: RoutedDrawerProps["PaperProps"] = {
   },
 }
 
+const CapturePageView: React.FC<{
+  resourceId: number
+  open: boolean
+}> = (props) => {
+  const { resourceId, open } = props
+  const resource = useLearningResourcesDetail(Number(resourceId))
+  const posthog = usePostHog()
+
+  if (APP_SETTINGS.posthog?.enabled && resource.isSuccess && open) {
+    posthog.capture("lrd_view", {
+      resourceId: resourceId || "unknown",
+      readableId: resource.data.readable_id,
+      platformCode: resource.data.platform?.code,
+      resourceType: resource.data.resource_type,
+    })
+  }
```

Comment:
We should wrap this in `useEffect`:
```ts
useEffect(() => {
  if (APP_SETTINGS.posthog?.enabled && resource.isSuccess && open) { ... }
}, [id])
```
otherwise:
- **important**  `posthog.capture` will be called on every render, even if the `id` hasn't changed. 
- **theoretical** React has been saying for a long time that side-effects should always be in `useEffect`, not in the render function. I believe that's because there's a planned feature ("concurrent mode") where react may start rendering something and abort the render in order to render some other, higher priority component. The idea is that `useEffect` would only be called when the component finishes rendering

**Aside:** I would just do this as a hook, rather than a component.

If you keep it as a component, you can return `null` rather than the empty fragment.

Reply:
@jkachel It's possible that nothing right now is triggering extra renders of the drawer, but we shouldn't count on it. And with the above `useEffect`, you might even remove the `open` dependency: The drawer content probably does re-render during closing, which maybe is why the `open` check got added in first place.

Reply:
I refactored this to be a hook in https://github.com/mitodl/mit-open/pull/746/commits/5e9394480c6c837df4732a11855fa79f57134e50 .